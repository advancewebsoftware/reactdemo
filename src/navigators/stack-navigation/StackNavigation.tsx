import React from 'react';
import {HomeScreen, WebViewScreen} from '../../modules';
import {NavigationContainer} from '@react-navigation/native';
import {AppStackNavigatorProps, RoutePage} from '../../types';
import {TransitionPresets, createStackNavigator} from '@react-navigation/stack';

const Stack = createStackNavigator<AppStackNavigatorProps>();

// Main Navigation component

const StackNavigation: React.FC = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator screenOptions={{headerShown: false}}>
        <Stack.Screen name={RoutePage.Home} component={HomeScreen} />
        <Stack.Screen
          name={RoutePage.WebView}
          component={WebViewScreen}
          options={{
            headerShown: false,
            gestureEnabled: true,
            ...TransitionPresets.ModalFadeTransition,
          }}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default StackNavigation;
