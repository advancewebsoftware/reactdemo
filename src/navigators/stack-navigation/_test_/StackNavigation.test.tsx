import {render} from '@testing-library/react-native';
import React from 'react';
import StackNavigation from '../StackNavigation';

jest.mock('@react-navigation/stack', () => ({
  createStackNavigator: () => ({
    Navigator: 'mocked-Navigator',
    Screen: 'mocked-Screen',
  }),
  TransitionPresets: jest.fn(),
}));

jest.mock('react-redux', () => ({
  Provider: jest.fn(),
}));

jest.mock('react-native-webview', () => {});

const component = <StackNavigation />;

describe('StackNavigate', () => {
  it('renders correctly', () => {
    const container = render(component);

    expect(container.toJSON()).toMatchSnapshot();
  });
});
