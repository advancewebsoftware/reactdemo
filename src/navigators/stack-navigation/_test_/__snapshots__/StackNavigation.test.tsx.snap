// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackNavigate renders correctly 1`] = `
<mocked-Navigator
  screenOptions={
    {
      "headerShown": false,
    }
  }
>
  <mocked-Screen
    component={[Function]}
    name="Home"
  />
  <mocked-Screen
    component={[Function]}
    name="WebView"
    options={
      {
        "gestureEnabled": true,
        "headerShown": false,
      }
    }
  />
</mocked-Navigator>
`;
